### PR TITLE
fix: Extend character limit for API, query, and datasource names

### PR DESCRIPTION
**Pull request by saiprabhu-dandanayak**
### Bug : [Bug Link](https://github.com/appsmithorg/appsmith/issues/6489)
 I have raised this PR inorder to `To extends the character limit for API, query, and datasource names, allowing for more descriptive and clearer identifiers in the system.`

#### here we gave query name more than 30 characters
image

#### When you hover over a query that isn't visible, a tooltip appears displaying the query's name as below.
querynamesimage

## Summary by CodeRabbit

- **New Features**
  - Input fields across various components now support a maximum length of 256 characters and utilize full width for better user experience.

- **Enhancements**
  - Improved name validation logic for entity names to simplify and enhance performance.
  - Added special character removal for certain input fields to ensure cleaner data entries.

- **Bug Fixes**
  - Fixed potential issues with callback dependencies to ensure more accurate updates and functionality.

### CODE REVIEW

File: app/client/src/components/editorComponents/ActionNameEditor.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { memo } from "react";
+import React, {memo} from "react";
```
Comment (ayushpahwa) [inline app/client/src/components/editorComponents/ActionNameEditor.tsx:1]: Is this a prettier miss?

**Review by ayushpahwa — CHANGES_REQUESTED** [review]
No description for changes or a connected issue and the prettier seems to be failing.

**Comment by saiprabhu-dandanayak** [comment]
> No description for changes or a connected issue and the prettier seems to be failing.

Hi @ayushpahwa , added description along with the bug link.

File: app/client/src/components/editorComponents/ActionNameEditor.tsx (hunk 1 above)
Comment (saiprabhu-dandanayak) [reply app/client/src/components/editorComponents/ActionNameEditor.tsx]: fixed.

**Comment by saiprabhu-dandanayak** [comment]
> > No description for changes or a connected issue and the prettier seems to be failing.
> 
> Hi @ayushpahwa , added description along with the bug link.

Hi @ayushpahwa , I've just pulled the latest changes from the release branch , fixed prettier linting issue in the current one , raised an updated pull request (PR) to reflect these changes , please review if you have enough bandwidth , please let me know your feedback.

File: app/client/src/components/utils/NameEditorComponent.tsx [hunk 2]
```diff
@@ -127,7 +124,7 @@ function NameEditor(props: NameEditorProps) {
         dispatch(dispatchAction({ id: entityId, name }));
       }
     },
-    [dispatch, isInvalidNameForEntity, entityId, entityName],
+    [hasActionNameConflict, entityName, suffixErrorMessage],
```
Comment (ayushpahwa) [inline app/client/src/components/utils/NameEditorComponent.tsx:127]: Whats the reason to remove the dependencies here?

File: app/client/src/utils/helpers.tsx [hunk 3]
```diff
@@ -172,13 +172,9 @@ function removeClass(ele: HTMLElement, cls: string) {
...
 
-export const removeSpecialChars = (value: string, limit?: number) => {
-  const separatorRegex = /\W+/;
-  return value
-    .split(separatorRegex)
-    .join("_")
-    .slice(0, limit || 30);
-};
+export const removeSpecialChars = (value: string) => {
+  return value.replace(/\W+/g, '_');
```
Comment (ayushpahwa) [inline app/client/src/utils/helpers.tsx:176]: Why remove the separator regex here?

**Review by ayushpahwa — CHANGES_REQUESTED** [review]
Thanks for the fixes. Apologies for the delay, I was OOO. I've re-review the PR. Left 2 comments. Also, this might fail tests, I'm triggering the CI. @rohan-arthur can you also help if this is going to affect any ongoing projects? Thanks!

File: app/client/src/utils/helpers.tsx (hunk 3 above)
Comment (saiprabhu-dandanayak) [reply app/client/src/utils/helpers.tsx:176]: i haven't removed it , it was suggested by coderabbitai  to write in a more readable way , the functionality is being the same , but removed the slicing upto 30 characters.

**Comment by saiprabhu-dandanayak** [comment]
> Thanks for the fixes. Apologies for the delay, I was OOO. I've re-review the PR. Left 2 comments. On top of that, please add some unit tests and this might fail some existing tests (I'm triggering the CI).
> 
> @rohan-arthur can you also help if this is going to affect any ongoing projects? Thanks!

ok @ayushpahwa , will add unit test for the current changes done.

**Comment by hetunandu** [comment]
We have not made any decision on issue and no assignment was made for the same. A character limit of 256 is too high and needs more thought. Closing this PR

**Pull request closed without merging**